vip-api/src/main/java/fr/insalyon/creatis/vip/api/security/oidc/OidcResolver.java

```diff
@@ -0,0 +1,123 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class OidcResolver {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ConfigurationBusiness configurationBusiness;
+    private final OidcConfig oidcConfig;
+
+    @Autowired
+    public OidcResolver(ConfigurationBusiness configurationBusiness, OidcConfig oidcConfig) {
+        this.configurationBusiness = configurationBusiness;
+        this.oidcConfig = oidcConfig;
+    }
+
+    // Common "Bad credentials" exception for Jwt to User resolution errors
+    private BadCredentialsException authError() {
+        return new BadCredentialsException(
+                SpringSecurityMessageSource.getAccessor().getMessage(
+                        "AbstractUserDetailsAuthenticationProvider.badCredentials",
+                        "Bad credentials"));
+    }
+
+    // Get DB User from authenticated JWT, using email-based resolution
+    private User getVipUser(Jwt jwt) throws BadCredentialsException {
+        String email = jwt.getClaim("email");
+        if (email == null) { // no email field in token
+            logger.info("Can't authenticate from OIDC token: no email in token");
+            throw authError();
+        }
+        User vipUser;
+        try {
+            vipUser = configurationBusiness.getUserWithGroups(email);
+        } catch (BusinessException e) { // DB lookup failed
+            logger.error("Error when getting user from OIDC token: doing as if there is an auth error", e);
+            throw authError();
+        }
+        if (vipUser == null) { // user not found
+            logger.info("Can't authenticate from OIDC token: user does not exist in VIP: {}", email);
+            throw authError();
+        }
+        if (vipUser.isAccountLocked()) { // account locked
+            logger.info("Can't authenticate from OIDC token: account is locked: {}", email);
+            throw authError();
+        }
+        return vipUser;
+    }
+
+    // Create authorities list from jwt claims.
+    // Using resource_access.<resource_name>.roles is Keycloak-specific.
+    private List<GrantedAuthority> parseAuthorities(User user, Jwt jwt) {
+        List<String> roles = new ArrayList<>(); // default to no roles
+        try {
+            String resource = oidcConfig.getIssuerResourceName(jwt.getIssuer().toString());
+            Map<String, Object> resourceAccess = jwt.getClaimAsMap("resource_access");
+            Map<String, Object> realmAccess = (Map<String, Object>) resourceAccess.get(resource);
+            roles = (List<String>) realmAccess.get("roles");
+            // here we could also map an authority from user level, as done by Apikey auth:
+            // roles.add("ROLE_" + user.getLevel().name().toUpperCase());
+            // but the existing Keycloak only used JWT-provided authorities
+        } catch (Exception e) {
```

Comment:
Again avoid all Exception catching.
Handle only checked exceptions and avoid runtime exception (like NPE) from happening.
Normally unexpected runtime exception happening during auth are managed and displayed properly.
Also we have the choice to silence the error (like you did), or throw a new exception (like a spring one).